Prototypes/BeleensMerge.cs

```diff
@@ -0,0 +1,497 @@
+//cs_include Scripts/CoreBots.cs
+//cs_include Scripts/CoreFarms.cs
+//cs_include Scripts/CoreStory.cs
+//cs_include Scripts/CoreAdvanced.cs
+//cs_include Scripts/Legion/CoreLegion.cs
+//cs_include Scripts/Legion/LegionExcercise/LegionExercise3.cs
+//cs_include Scripts/Legion/LegionExcercise/LegionExercise4.cs
+//cs_include Scripts/Nation/CoreNation.cs
+//cs_include Scripts/Nation/Various/TarosManslayer.cs
+//cs_include Scripts/Nation/Various/PurifiedClaymoreOfDestiny.cs
+//cs_include Scripts/Story/ArtixWedding.cs
+
+using Skua.Core.Interfaces;
+using Skua.Core.Models.Items;
+using Skua.Core.Options;
+
+public class BeleensMerge
+{
+    public IScriptInterface Bot => IScriptInterface.Instance;
+    public CoreBots Core => CoreBots.Instance;
+    public CoreFarms Farm = new();
+    public CoreStory Story = new();
+    public CoreAdvanced Adv = new();
+    public static CoreAdvanced sAdv = new();
+    public CoreLegion CoreLegion = new();
+    public LegionExercise3 LegionExercise3 = new();
+    public LegionExercise4 LegionExercise4 = new();
+    public CoreNation CoreNation = new();
+    public TarosManslayer TarosManslayer = new();
+    public ArtixWedding ArtixWedding = new();
+
+    public List<IOption> Generic = sAdv.MergeOptions;
+    public string[] MultiOptions = { "Generic", "Select" };
+    public string OptionsStorage = sAdv.OptionsStorage;
+    // [Can Change] This should only be changed by the author.
+    //              If true, it will not stop the script if the default case triggers and the user chose to only get mats
+    private bool dontStopMissingIng = false;
+
+    public void ScriptMain(IScriptInterface bot)
+    {
+        Core.SetOptions();
+
+        BuyAllMerge();
+
+        Core.SetOptions(false);
+    }
+
+    public void BuyAllMerge()
+    {
+        //Only edit the map and shopID here
+        Adv.StartBuyAllMerge("tower", 347, findIngredients);
+
+        #region Dont edit this part
+        void findIngredients()
+        {
+            ItemBase req = Adv.externalItem;
+            int quant = Adv.externalQuant;
+            int currentQuant = req.Temp ? Bot.TempInv.GetQuantity(req.Name) : Bot.Inventory.GetQuantity(req.Name);
+            if (req == null)
+            {
+                Core.Logger("req is NULL");
+                return;
+            }
+
+            switch (req.Name)
+            {
+                default:
+                    bool shouldStop = Adv.matsOnly ? !dontStopMissingIng : true;
+                    Core.Logger($"The bot hasn't been taught how to get {req.Name}." + (shouldStop ? " Please report the issue." : " Skipping"), messageBox: shouldStop, stopBot: shouldStop);
+                    break;
+                #endregion
+
+                case "Platinum Wings":
+                    if (!Core.IsMember)
```

Comment:
for things that require a membership you could just wrap them in a
if (Core.IsMember)
{
stuff
}
🤷  if the player isn't a mem it'll just skip it 👏 -- check the rest as well please